.github/workflows/publish.yml

```diff
@@ -0,0 +1,53 @@
+name: publish
+
+on:
+  pull_request: # TODO: remove this before merging, adding for testing purposes
+  push:
+    branches:
+      - master
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      # temp allows us to run on this PR
+      - name: Checkout repo
+        uses: actions/checkout@master
+        with:
+          ref: 'release/v1.0.0'
+
+      # replace above with this before merging, when removing run on PR
+      # - name: checkout
+      #   uses: actions/checkout@v1
+
+      - name: Configure CI Git User
+        run: |
+          git config --global user.name '@tteltrab'
+          git config --global user.email 'tteltrab@users.noreply.github.com'
+          git remote set-url origin https://$GITHUB_ACTOR:$GITHUB_PAT@github.com/IBM/slack-wrench.git
+        env:
+          GITHUB_PAT: ${{ secrets.GITHUB_PAT }}
+
+      - name: Install Packages
+        run: yarn install --frozen-lockfile
+
+      - name: Test
+        run: yarn test
```

Comment:
since this happens on push, I think it makes sense to run tests? _theoretically_ a package could change between PR and the publish build on master that breaks us.

Reply:
I agree that we should run tests, I just worry that there's now two different ways that we run tests in workflows, why not one? 

This kinda goes back to my question a while ago on why have many workflows with a single job vs having one workflow with many jobs? Functionally they'd run the same correct?

That way we could have something like:

```yaml
jobs:
  unit:
  lint:
  publish:
    needs: [ unit, lint ]
    if: github.event =='push' && github.ref == 'master'
```

[`if` docs](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idif)
[issue](https://github.community/t5/GitHub-Actions/Github-Actions-CI-CD-Need-optional-jobs/td-p/29696#)

Reply:
FWIW, if the answer is "Oh yeah, maybe we should do that" lets refactor that in a different PR just to keep this simple.

Reply:
Yeah, last we chatted about this we'd mentioned we'd implement a few of these and then improve/refactor once we decided on a best approach, which I don't think we have yet? I'm potentially okay with this approach.

Also not sure they'd run exactly the same. I know this works, so let's say "probably" and refactor in another PR.